sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,252 @@
+            "test_id": test_id,
+            "nodes": [_get_runner_results(runner) for runner in self._benchmark_runners]}
+
+        LOGGER.info("Benchmark results doc is:\n%s", results)
+        self._es.create_doc(index=ES_INDEX, doc_type=None, doc_id=test_id, body=results)
+        return results
+
+    def print_benchmark_results_to_log(self):
+        results = self._collect_benchmark_output()
+        LOGGER.info("Benchmark results:")
+        LOGGER.info(results)
+
+
+class ScyllaNodeBenchmarkRunner:
+    """
+    ScyllaNodeBenchmarkRunner installs and runs benchmarking
+    tools on given cluster nodes and collects the output.
+    """
+
+    def __init__(self, node: "BaseNode"):
+        self._node = node
+        self._remoter: RemoteCmdRunnerBase = node.remoter
+        self.node_instance_type = self._get_db_node_instance_type()
+        self.benchmark_results = {"sysbench": {}, "cassandra-fio": {}}
+
+    def install_benchmark_tools(self) -> bool:
+        # pylint: disable=broad-except
+        try:
+            self._clone_repo("https://github.com/akopytov/sysbench.git")
+            self._clone_repo("https://github.com/KnifeyMoloko/cassandra-fio.git")
+            self._install_ubuntu_prerequisites()
+            self._build_and_install_sysbench()
+        except Exception as exc:
+            LOGGER.warning("Installing benchmark tools failed with exception:\n%s", exc)
+        return self._is_sysbench_installed()
+
+    @property
+    def node_name(self):
+        return self._node.name
+
+    def _get_db_node_instance_type(self):
+        backend = self._node.parent_cluster.params.get("cluster_backend")
+        if backend == "aws":
+            return self._node.parent_cluster.params.get("instance_type_db")
+        elif backend == "gce":
+            return self._node.parent_cluster.params.get("gce_instance_type_db")
+        else:
+            LOGGER.warning("Unrecognized backend type, defaulting to 'Unknown' for"
+                           "db instance type.")
+            return None
+
+    def _install_ubuntu_prerequisites(self) -> bool:
+        # pylint: disable=broad-except
+        package_list = ["make", "automake", "libtool", "pkg-config", "libaio-dev", "fio"]
+        try:
+            LOGGER.info("Installing Ubuntu prerequisites for the node benchmarks...")
+            for pkg in package_list:
+                self._node.install_package(pkg, wait_for_package_manager=False)
+            return True
+        except Exception as exc:
+            LOGGER.warning("Failed to install Ubuntu prerequisites for the node benchmarking tools. "
+                           "Exception:\n%s", exc)
+            return False
+
+    def _build_and_install_sysbench(self) -> bool:
+        # pylint: disable=broad-except
+        build_and_install_script = """\
+            cd ./sysbench
+            ./autogen.sh
+            ./configure --without-mysql
+            make -j
+            make install
```

Comment:
Do we really want to build it on each separate test run? Maybe we want to reuse it from S3? Picking up latest version we put there?
Even if not add it now, but at least, add a "TODO" comment and create new task in our task tracker.
What do you think?

Reply:
It sounds good in theory. My fear is that: a) downloading from S3 might take longer than the current process b) we're adding maintenance cost by this. That being said, I created a Trello taks for this: https://trello.com/c/KIbuX25o and we'll see how it goes.

Reply:
@aleksbykov docker isn't installed on DB nodes (installing docker would take more time)

@KnifeyMoloko why not taking the official binary of it ? (instead of building it)

https://packagecloud.io/akopytov/sysbench/install#bash-deb